dev-lang/vala/vala38-0.38.8.recipe

```diff
@@ -0,0 +1,131 @@
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala38-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersionCompat
+	cmd:vala$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valadoc$secondaryArchSuffix
+	cmd:vapicheck$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_${portVersion%.*}$secondaryArchSuffix
+	cmd:vala_gen_introspect_${portVersion%.*}$secondaryArchSuffix
+	cmd:valac_${portVersion%.*}$secondaryArchSuffix
+	cmd:valadoc_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapicheck_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapigen_${portVersion%.*}$secondaryArchSuffix
+	lib:libvala_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	lib:libvaladoc_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libdbus_1$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgvc$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+REPLACES="
+	vala${secondaryArchSuffix}_common
+	"
+
+PROVIDES_devel="
+	vala${secondaryArchSuffix}_devel = $portVersionCompat
+	"
+REQUIRES_devel="
+	vala$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcgraph$secondaryArchSuffix
+	devel:libdbus_1$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgvc$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:autoreconf
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:diff
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage vala$secondaryArchSuffix \
+	$libDir/libvala-${portVersion%.*}.so.$libVersion \
+	$libDir/libvaladoc-${portVersion%.*}.so.$libVersion \
+	$binDir/vala \
+	$binDir/valac \
+	$binDir/valadoc \
+	$binDir/vapicheck \
+	$binDir/vapigen \
+	$binDir/vala-${portVersion%.*} \
+	$binDir/valac-${portVersion%.*} \
+	$binDir/valadoc-${portVersion%.*} \
+	$binDir/vapicheck-${portVersion%.*} \
+	$binDir/vapigen-${portVersion%.*}
+
+PATCH()
+{
+	sed -i -e 's/-lm/-lroot/' tests/testrunner.sh
+}
+
+BUILD()
+{
+	autoreconf -fi
+	runConfigure ./configure
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	rm -f $libDir/*.la
+	rm -f $libDir/*/*.la
+	rm -f $libDir/*/*/*/*.la
+
+	fixPkgconfig
+
+	#devel and common packages
+	packageEntries devel \
+		$developDir
+}
+
+TEST()
+{
+	LIBRARY_PATH="$sourceDir/libvaladoc/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
+	LIBRARY_PATH="$sourceDir/vala/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
```

Comment:
I'm not sure this would work. What do you think of the following variant?
```diff
 TEST()
 {
-	LIBRARY_PATH="$sourceDir/libvaladoc/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
-	LIBRARY_PATH="$sourceDir/vala/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
+ 	libvalaDir="$sourceDir"/vala/.libs
+ 	libvaladocDir="$sourceDir"/libvaladoc/.libs
+	LIBRARY_PATH="$libvalaDir:$libvaladocDir${LIBRARY_PATH:+:$LIBRARY_PATH}" \
 	make check
 }
```

Reply:
It worked just now, but I can check the variant :)

Reply:
Both seem to be ok (for secondaryArch on x86_gcc2), if you prefer to use your variant I'm glad to update it to that one ... but maybe someone could check it also on x86_64? ...

Reply:
I've built graphviz on x86_64 (a few days ago) but it fails to build on x86 second arch, so I can't try to build vala-0.38.8 on x86 second arch.
Anyway, when I try to build vala-0.38.8 on x86_64 I get this:
```
checking for glib-2.0 >= 2.40.0 gobject-2.0 >= 2.40.0... yes
checking for gmodule-2.0 >= 2.40.0... yes
checking for libgvc >= 2.16... yes
checking for CGRAPH... no
...
make  all-am
make[3]: Entering directory '/sources/vala-0.38.8/libvaladoc'
  CCLD     libvaladoc-0.38.la
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/5.4.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lgvc
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/5.4.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lcgraph
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/5.4.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lcdt
collect2: error: ld returned 1 exit status
Makefile:992: recipe for target 'libvaladoc-0.38.la' failed
make[3]: *** [libvaladoc-0.38.la] Error 1
```
I can't investigate at the moment but I'll try to do so soon.

Reply:
Regarding the `LIBRARY_PATH` in `TEST()`, you are right. Both work.
But you might wish to simplify your code a little:
```
	LIBRARY_PATH="$sourceDir/libvaladoc/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
	LIBRARY_PATH="$sourceDir/vala/.libs:$LIBRARY_PATH" \
 	make check
```
You can also keep what you have right now, but there are 2 small downsides:
1) it is not obvious to the reader that `LIBRARY_PATH` will have `vala/.libs` before `libvaladoc/.libs`.
2) it is not very useful to have the `${LIBRARY_PATH:+:$LIBRARY_PATH}` in each line, as it is only needed in the first line.

For these reasons I would recommend using the variant I suggested above, because it's easier to understand what it does. Also, it only has one `${LIBRARY_PATH:+:$LIBRARY_PATH}`.

There are other variants we could use:
```
	LIBRARY_PATH="$sourceDir/vala/.libs:$sourceDir/libvaladoc/.libs${LIBRARY_PATH:+:$LIBRARY_PATH}" \
	make check
```
Pros: we don't need to define valaDir and libvaladocDir.
Cons: the line is around 100 chars.
But I think it would be OK, here, to have a line > 80 chars.
So this is what I'd recommend, after all, if nobody objects having a long line here.

And here's a final variant:
```
	LIBRARY_PATH="$sourceDir/vala/.libs:$sourceDir/libvaladoc/.libs\
${LIBRARY_PATH:+:$LIBRARY_PATH}" \
	make check
```
It's ugly but at least it fits in less than 80 chars.

Reply:
It requires graphviz (hence my last comment here - and I've posted about that for the previous version for vala) :)
Just build graphviz 2.40.1 on x86_64 and then vala 0.38.8, both build ok ... (just need to run the test for vala, but I don't think that one will be a problem)

Reply:
Tests run ok on x86_gcc2 (for x86) and x86_64 ...

Reply:
That's what I did. I have even installed graphviz 2.40.1 but for some reason unknown to me I  vala 0.38.8 fails to build.

The tests for vala 0.36.12 are all OK on all architectures. The only change we might have to do is moving some files to the devel package.

Reply:
I'll wait until someone steps up here to put it in the right sentence before pushing an update :)